sourcegraph/cody
# feat: add "regenerate" code lens for edits

part of https://github.com/sourcegraph/cody/issues/1359

feat: add retry code lens for fixup tasks

This adds a new "regenerate" code lens option for edit/fixup tasks. When clicked, it will try to regenerate a response using the same fixup instruction.

Changes:

- Register new "cody.fixup.codelens.regenerate" command 
- Add getRegenerateLens() to generate`regenerate` code lens
- In getLensesForTask(), include`regenerate` lens in returned array
- Add retry() method to handle retry click
  - Discard previous task and code lens
  - Executes edit-code command with same instructions to create a new task for regenerating code

## Test plan



1. Select a section of code
2. Run /edit command: `/edit add comments inline`
3. Check the diff to see what was edited
4. Click on the `Retry` code lens again
5. Check diff again, you should see the comments were updated by Cody again

![image](https://github.com/sourcegraph/cody/assets/68532117/2f805ee2-481c-4fcf-8533-d832bf78bc51)

## Demo

https://github.com/sourcegraph/cody/assets/68532117/62c53c0a-d777-46e1-93e2-fb228912fafe



This is expected to work the same when Tom has made the changes so that all Fixup tasks are applied automatically by default.
